src/js/api/posts/create.mjs

```
@@ -0,0 +1,32 @@
+import { API_POSTS, API_SOCIAL_URL } from "../constants.mjs";
+import { authFetch } from "../authFetch.mjs";
+import { load } from "../../storage/index.mjs";
+
+const method = "post";
+const author = "?_author=true";
+//const action = "/posts";
+
+export async function createPost(postData) {
+    const createUrl = `${API_SOCIAL_URL}${API_POSTS}${author}`;
+    try {
+        const profileData = load("profile").name;
+        const postDataWithAuthor = { ...postData, author: profileData };
+        const response = await authFetch(createUrl, {
+            method,
+            body: JSON.stringify(postDataWithAuthor)
+        });
+        if (response.ok) {
+            const createdPost = await response.json();
+            let posts = JSON.parse(localStorage.getItem("posts")) || [];
+            posts.push(createdPost)
+            localStorage.setItem("posts", JSON.stringify(posts));
+            window.location.href = "/profile/";
+            return await createdPost;                 
+        } else {
+            throw new Error ("Failed to create post");                          
+        }
+    } catch (error) {
+        console.error("Failed to create post")
+        throw error;
```

`catch block`: to suppress an error
`throw`: to create an error

Understand the differences, they have different purposes. Here you're basically catching an error and throwing it again.

**Suggestion:** remove the throw from the catch block and use the `try/catch` block in a UI/UX code block e.g event listener.